app/services/job_manager.py

```
@@ -1786,6 +1790,32 @@ async def _process_job(self, job_id: str):
                     raise ConversionCancelled("Conversion cancelled")
 
+            if job.input_kind == InputKind.DIRECTORY:
+                # Re-check directory-input safety after acquiring the source
+                # subtree lock and immediately before invoking the native
+                # recursive packer. A queued PS3 folder job may have been
+                # valid when planned but mutated before it starts; this guards
+                # the exact tree makeps3iso is about to read. Run it *before*
+                # clearing any existing output so a safety rejection on an
+                # overwrite job is non-destructive: the user's prior ISO/split
+                # set is only removed once the source is confirmed still safe.
+                if not await run_in_threadpool(is_safe_directory_tree, input_path):
```

Ruff (0.15.20)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)